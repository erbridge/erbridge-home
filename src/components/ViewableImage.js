import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component, Fragment } from 'react';
import Img from 'react-image';

import Lightbox from './Lightbox';

import './ViewableImage.scss';

class ViewableImage extends Component {
  static propTypes = {
    alt: PropTypes.string.isRequired,
    className: PropTypes.string,
    src: PropTypes.string.isRequired,
  };

  state = {
    lightboxIsOpen: false,
  };

  openLightbox() {
    this.setState({ lightboxIsOpen: true });
  }

  closeLightbox() {
    this.setState({ lightboxIsOpen: false });
  }

  render() {
    const { alt, className, src } = this.props;
    const { lightboxIsOpen } = this.state;

    return (
      <Fragment>
        <Img
          className={classnames(className, 'ViewableImage')}
          src={src}
          alt={alt}
          unloader={
            <span
              className={classnames(className, 'ViewableImage__placeholder')}
            >
              {alt}
            </span>
          }
          onClick={() => this.openLightbox()}
        />
        <Lightbox
          src={src}
          isOpen={lightboxIsOpen}
          onClose={() => this.closeLightbox()}
        />
      </Fragment>
    );
  }
}

export default ViewableImage;
