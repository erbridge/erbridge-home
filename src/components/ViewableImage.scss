.ViewableImage {
  display: flex;
  justify-content: center;

  img {
    cursor: pointer;
  }
}
