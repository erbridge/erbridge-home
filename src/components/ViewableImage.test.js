import { shallow } from 'enzyme';
import React from 'react';

import ViewableImage from './ViewableImage';

it('renders correctly', () => {
  expect(
    shallow(
      <ViewableImage className="Test__image" src="/test.jpg" alt="test" />,
    ),
  ).toMatchSnapshot();
});
