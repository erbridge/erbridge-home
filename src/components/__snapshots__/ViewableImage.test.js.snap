// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Img
    alt="test"
    className="ViewableImage"
    container={[Function]}
    decode={true}
    loader={false}
    onClick={[Function]}
    src="/test.jpg"
    unloader={
      <span
        className="ViewableImage__placeholder"
      >
        test
      </span>
    }
  />
  <Lightbox
    isOpen={false}
    onClose={[Function]}
    src="/test.jpg"
  />
</Fragment>
`;
